stream/src/public_key/nonce.rs

```diff
@@ -3,52 +3,48 @@ use chacha20poly1305::Nonce;
 
 /// A struct that holds the nonce information.
 ///
-/// The nonce contains:
-/// - `dialer`: a boolean that is true if the nonce is for the dialer side.
-/// - `iter` and `seq`: combined, an 80-bit value that can be incremented.
+/// Holds a `u128` value that is incremented by 2 each time the nonce is incremented.
+/// The least-significant bit is used to indicate the dialer side and does not change.
+///
+/// Is able to be incremented up-to 96 bits (12 bytes) before overflowing.
 pub struct Info {
-    dialer: bool,
-    iter: u16,
-    seq: u64,
+    counter: u128,
 }
 
+/// If the counter is greater-than-or-equal to this value, it is considered to have overflowed.
+/// This is 2^96 or, in binary, one followed by 96 zeros.
+const OVERFLOW_VALUE: u128 = 1 << 96;
+
 impl Info {
     pub fn new(dialer: bool) -> Self {
         Self {
-            dialer,
-            iter: 0,
-            seq: 0,
+            counter: if dialer { 1 } else { 0 },
         }
     }
 
     /// Increments the nonce.
     ///
-    /// `seq` holds the least significant 64 bits of the nonce, and `iter` holds the most significant 16 bits.
-    /// An error is returned if-and-only-if the nonce overflows.
+    /// Counter is incremented by 2, maintaining the dialer bit.
```

Comment:
I'm not sure "maintaining the dialer bit" is the right way to say this.

I think it should be "preventing nonce conflict between connected parties"